### PR TITLE
Tidy up the generation of postData by the client

### PR DESCRIPTION
**Pull request by deepakunni3**
The generation of postData JSON by `AnnotTrack.js` and `SequenceTrack.js` was inconsistent. 
In some places it was a JSON object which was stringified where as in other places it was just building strings that represented JSON. 

This inconsistency led to improperly formatted JSON being sent to the server.

This PR aims at standardizing this process.

### CODE REVIEW

**Comment by deepakunni3** [comment]
@nathandunn Read for test and review.

PS. Try all operations to make sure that nothing was affected in light of this fix.

**Comment by nathandunn** [comment]
The changes all looked great except where noted.  Testing now.

**Comment by nathandunn** [comment]
@deepakunni3  Deleting annotations doesn't quite work.   I have some fixes that seem to fix it and removed some other junk.

**Comment by nathandunn** [comment]
But retest.

File: client/apollo/js/View/Track/AnnotTrack.js [hunk 1]
```diff
@@ -6425,9 +6425,7 @@ define([
 
             executeUpdateOperation: function (postData, loadCallback) {
                 if(!postData.hasOwnProperty('clientToken')){
-                    var postObject = JSON.parse(postData);
-                    postObject.clientToken = this.getClientToken();
-                    postData = JSON.stringify(postObject);
+                    postData.clientToken = this.getClientToken();
```
Comment (deepakunni3) [inline client/apollo/js/View/Track/AnnotTrack.js]: Good catch. Now that `postData` is strictly a JSON object, another `JSON.parse` will definitely fail. Which is what led to the error.

**Comment by nathandunn** [comment]
still wondering if this should be applied to the master branch as well as the websocket update may have broken a few other things.  I'm taking a look at #1665.   I'm wondering if it needs a related cleanup.

**Comment by deepakunni3** [comment]
@nathandunn Good catch. 

Retested and it works fine now.

**Comment by nathandunn** [comment]
Also needs to retest the sequence alteration with this fix.  I think . . . I'll double-check.

**Comment by monicacecilia** [comment]
Yep. Clean up AND re-testing needed. . .

**Comment by deepakunni3** [comment]
Sequence alterations work on this branch. But yes, lets carefully test all operations before we merge this in.

**Comment by nathandunn** [comment]
@deepakunni3  Did my fix help or hurt it?    I removed the string call in favor of an object call, but I was testing the wrong branch.

**Comment by deepakunni3** [comment]
@nathandunn The fix you pushed to `https://github.com/deepakunni3/Apollo/tree/tidy_up_operations` does get rid of the error you saw (and even I did) while deleting an annotation.

**Comment by nathandunn** [comment]
@deepakunni3  This fix:  https://github.com/GMOD/Apollo/pull/1664/commits/932b6ac7e5b94e40ac8e514e427084698fe2c04c

?

I know the first one worked.  Was just wondering about this one.

**Comment by deepakunni3** [comment]
@nathandunn Sorry, I just saw that right now.

That commit had no effect on this branch. Because in either way, it works just fine.

**Comment by deepakunni3** [comment]
@nathandunn This is ready to re-test again.

I made sure that only AnnotTrack's `getUniqueTrackName()` method is being called, even in SequenceTrack. Also made sure that the proper JSON formatted sequenceList is returned by `getUniqueTrackName()`.

**Comment by nathandunn** [comment]
Looks good!

**Comment by nathandunn** [comment]
And I tested it and it worked.

**Comment by deepakunni3** [comment]
awesome!

**Pull request merged**